firefox-ios/Client/Frontend/Browser/TopTabsViewController.swift

```diff
@@ -343,6 +343,7 @@ extension TopTabsViewController: TopTabCellDelegate {
 
 extension TopTabsViewController: PrivateModeUI {
     func applyUIMode(isPrivate: Bool, theme: Theme) {
+        // TODO: Ideally we shouldn't be creating new tabs as a side-effect of updating UI theme. Investigate refactor.
```

Comment:
agreed! we can prob investigate in removing `applyUIMode` now that we have `applyTheme`

Reply:
@mattreaganmozilla Can you create a ticket to follow up and refactor this and add the ticket number to the TODO?
@cyndichin Even applyTheme is not the correct place for this. Whenever you have time to pick up the refactor ticket Matt creates feel free to drop some time in my calander and I'll talk you through 1) this issues with this approach and 2) a better way to handle this. 
This will bite us in the ass again if we leave it in the code long term.

Reply:
sounds great, thank you!

Reply:
> Can you create a ticket to follow up and refactor this and add the ticket number to the TODO?

@OrlaM Sure, filed: https://mozilla-hub.atlassian.net/browse/FXIOS-8907, I'll add the ticket to the ToDo